File: 6kyu/Multiples of 3 or 5/Solution.java
// https://www.codewars.com/kata/514b92a657cdc65150000006/solutions/java

public class Solution {

  public int solution(int number) {
    int sum = 0;
    
    for (int i = 1; i < number; i += 1) {
      int this_number = number - i;
      
      if (this_number % 3 == 0 || this_number % 5 == 0) {
        sum += this_number;
      }
    }
    
    return sum;
  }
}

File: 6kyu/Multiples of 3 or 5/solution.kt
// https://www.codewars.com/kata/514b92a657cdc65150000006/solutions/kotlin

fun solution(number: Int): Int {
    var sum:Int = 0;
    
    for(x in 1..(number-1)) {
        sum += if (x % 3 === 0 || x % 5 === 0) x else 0;
    }
    
  return sum
}

// 함수형
// fun solution(number: Int) = (1 until number).filter { it % 3 == 0 || it % 5 == 0 }.sum()
